scripts/mellanox_buffer_migrator.py

```diff
@@ -303,64 +484,101 @@ def mlnx_migrate_buffer_profile(self, old_version, new_version):
-                    new_profile = new_lossless_profiles.get(name)
-                    if not default_profile or not new_profile:
-                        continue
-                    default_profile['dynamic_th'] = '0'
-                    default_profile['xoff'] = str(int(default_profile['size']) - int(default_profile['xon']))
-                    default_profile['pool'] = '[BUFFER_POOL|ingress_lossless_pool]'
-                    if profile == default_profile:
-                        default_profile['size'] = new_profile['size']
-                        default_profile['xon'] = new_profile['xon']
-                        default_profile['xoff'] = str(int(default_profile['size']) - int(default_profile['xon']))
+            default_headrooms_old = default_headroom_sets_old.get("spc1_headroom")
+            default_headrooms_new = default_headroom_sets_new.get("spc1_headroom")
+
+        if default_headrooms_old and default_headrooms_new:
+            # match the old lossless profiles?
+            for headroom_set_name, lossless_profiles in default_headrooms_old.iteritems():
+                if headroom_set_name == "mapping":
+                    continue
+                lossless_profiles = self.mlnx_migrate_get_headroom_profiles(lossless_profiles)
+                matched = True
+                for name, profile in configdb_buffer_profiles.iteritems():
+                    if name in lossless_profiles.keys():
+                        default_profile = self.mlnx_migrate_extend_headroom_profile(lossless_profiles.get(name))
+                        if profile != default_profile:
+                            log.log_info("Skip headroom profile set {} due to {} mismatched: {} vs {}".format(
+                                headroom_set_name, name, default_profile, profile))
+                            matched = False
+                            break
+                if matched:
+                    mapping = default_headroom_sets_new.get("mapping")
+                    if not mapping:
+                        new_headroom_set_name = headroom_set_name
+                        log.log_info("Migrate profile set {} ".format(headroom_set_name))
+                    else:
+                        new_headroom_set_name = mapping.get(headroom_set_name)
+                        if type(new_headroom_set_name) is tuple:
+                            log.log_info("Use headroom profiles map {}".format(mapping))
+                            maptype, mapping = new_headroom_set_name
+                            if maptype == "skumap":
+                                new_headroom_set_name = mapping.get(self.sku)
+                        if not new_headroom_set_name:
+                            new_headroom_set_name = headroom_set_name
+                    log.log_info("{} has been mapped to {} according to sku".format(headroom_set_name, new_headroom_set_name))
+                    break
+
+            if not matched:
```

Comment:
wouldn't this get overwritten for each iteration of the for loop in L521?

Reply:
Not really.
The logic is to iterate each flavor of the headroom sets and break if one matched. The `matched` will be set to true at the beginning of each iteration. It will be set to false and then `continue` if there is a difference between current db configuration and the flavor of default headroom info.
If the current db configuration matches none of the flavors, the `matched` will be false here.